src/drive/web/modules/services/components/Picker/Picker.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from 'react'
+import { IntentHeader, Button, withBreakpoints } from 'cozy-ui/react'
+import { ROOT_DIR_ID } from 'drive/constants/config'
+import Topbar from 'drive/web/modules/Layout/Topbar'
+import FileList from 'drive/web/modules/filelist/FileList'
+import {
+  Breadcrumb,
+  PreviousButton
+} from 'drive/web/modules/navigation/Breadcrumb'
+import { connect } from 'react-redux'
+import withReduxStore from './withReduxStore'
+import AddFolderButton from './AddFolderButton'
+import OriginHint from './OriginHint'
+import {
+  openFolder,
+  getOpenedFolderId,
+  getVisibleFiles
+} from 'drive/web/modules/navigation/duck'
+import styles from './picker.styl'
+
+class Picker extends Component {
+  state = {
+    path: []
+  }
+
+  updateBreadcrumb = folder => {
+    this.setState(state => {
+      const indexInPath = state.path.indexOf(folder)
+      const newPath =
+        indexInPath >= 0
+          ? state.path.slice(0, indexInPath + 1)
+          : [...state.path, folder]
+
+      return {
+        ...state,
+        path: newPath
+      }
+    })
+  }
+
+  navigateTo = folder => {
+    this.props.fetchFolder(folder.id)
+    this.updateBreadcrumb(folder)
+  }
+
+  goBack = () => {
+    const { path } = this.state
+    this.navigateTo(path[path.length - 2])
+  }
+
+  componentDidMount() {
+    const root = document.getElementById('main')
+    const data = root.dataset
+    this.headerIcon = data.cozyIconPath
```

Comment:
Another thing I'm not very happy about — but not sure where would be a good place to get this icon. Maybe passing it down as a prop from the `main.jsx` file?